src/.nuget/Microsoft.NET.Sdk.Optimization.ReadyToRun/sdk/Sdk.targets

```diff
@@ -0,0 +1,213 @@
+
+    <NETSdkError Condition="'$(RuntimeIdentifier)' ==''"
+                 ResourceName="RuntimeIdentifierWasNotSpecified"/>
+
+    <NETSdkError Condition="'$(TargetFramework)' ==''"
+                 ResourceName="AtLeastOneTargetFrameworkMustBeSpecified"/>
+
+    <!-- Determine whether the host OS platform-architecture supports compilation to the desired RuntimeIdentifier -->
+    <PropertyGroup>
+      <_TargetArchitecture>$(PlatformTarget)</_TargetArchitecture>
+      <_TargetArchitecture Condition="$(RuntimeIdentifier.EndsWith('arm64'))">arm64</_TargetArchitecture>
+
+      <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('windows'))">win</_OSPlatform>
+      <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('osx'))">osx</_OSPlatform>
+      <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('linux'))">linux</_OSPlatform>
+      <_OSArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)</_OSArchitecture>
+
+      <_UnsupportedReadyToRunCompilation>false</_UnsupportedReadyToRunCompilation>
+
+      <!-- First check that the host and target OS are the same -->
+      <_UnsupportedReadyToRunCompilation Condition="'$(_OSPlatform)' == 'win' and !$(RuntimeIdentifier.StartsWith('win'))">true</_UnsupportedReadyToRunCompilation>
+      <_UnsupportedReadyToRunCompilation Condition="'$(_OSPlatform)' == 'osx' and !$(RuntimeIdentifier.StartsWith('osx'))">true</_UnsupportedReadyToRunCompilation>
+
+      <!-- TODO: Allow cross-compilation to work for Crossgen in supported scenarios -->
+      <!-- Next check that the host can compile for the target OS architecture -->
+      <_UnsupportedReadyToRunCompilation Condition="'$(_OSArchitecture)' == 'x86' and ('$(_TargetArchitecture)' != 'x86' and '$(_TargetArchitecture)' != 'arm')">true</_UnsupportedReadyToRunCompilation>
+      <_UnsupportedReadyToRunCompilation Condition="'$(_OSArchitecture)' == 'arm' and '$(_TargetArchitecture)' != 'arm'">true</_UnsupportedReadyToRunCompilation>
+      <!-- TODO insert arm64 capabilities. Specifically, can we always assume arm64 architectures can emulate arm? What about x86? -->
+      <_UnsupportedReadyToRunCompilation Condition="'$(_OSArchitecture)' == 'arm64' and '$(_TargetArchitecture)' != 'arm64'">true</_UnsupportedReadyToRunCompilation>
+    </PropertyGroup>
+
+    <!-- TODO: Add this message as a resource string to the NET SDK so it can be localized -->
+    <NETSdkError Condition="'$(_UnsupportedReadyToRunCompilation)' == 'true'"
+        FormattedText="Creating ReadyToRun images is not supported for the RuntimeIdentifier $(RuntimeIdentifier) on the host OS $(_OSPlatform)-$(_OSArchitecture.ToLower())" />
+  </Target>
+
+  <Target Name="_InitializeReadyToRunCompilerProps"
+          Condition="'$(ReadyToRun)' == 'true'">
+    <!-- Get the coreclr path -->
+    <ItemGroup>
+      <!-- TODO: Switch to using %(ResolvedTargetingPack) to find the targeting pack-->
+      <_CoreclrResolvedPath Include="@(ResolvedAssembliesToPublish)"
```

Comment:
Is ReadyToRun supposed to work on Framework-dependent deployments? Because this seems to rely on self-contained to find coreclr.